ml-agents/mlagents/trainers/policy/policy.py

```diff
@@ -1,12 +1,151 @@
-from abc import ABC, abstractmethod
+from abc import abstractmethod
+from typing import Dict, List, Optional
+import numpy as np
 
 from mlagents_envs.base_env import DecisionSteps
+from mlagents_envs.exception import UnityException
+
 from mlagents.trainers.action_info import ActionInfo
+from mlagents_envs.base_env import BehaviorSpec
+from mlagents.trainers.settings import TrainerSettings, NetworkSettings
 
 
-class Policy(ABC):
-    @abstractmethod
+class UnityPolicyException(UnityException):
+    """
+    Related to errors with the Trainer.
+    """
+
+    pass
+
+
+class Policy:
+    def __init__(
+        self,
+        behavior_spec: BehaviorSpec,
+        seed: int,
```

Comment:
Just noticed something that we should change. The order of these params are different between `Policy` and `TFPolicy` - either order is fine as long as we're consistent.